bundles/org.eclipse.swt/Eclipse SWT Browser/common/org/eclipse/swt/browser/Browser.java

```diff
@@ -182,9 +182,9 @@ static int checkStyle(int style) {
 				if (current.equalsIgnoreCase ("webkit")) { //$NON-NLS-1$
 					DefaultType = SWT.WEBKIT;
 					break;
-				} else if (current.equalsIgnoreCase ("edge") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
-					DefaultType = SWT.EDGE;
 				} else if (current.equalsIgnoreCase ("ie") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
+					DefaultType = SWT.IE;
+				} else if (current.equalsIgnoreCase ("edge") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
This change breaks the following check (introduced by #1588) and therefore brings back the flickering:

https://github.com/amartya4256/eclipse.platform.swt/blob/7cee4b6bc7c8c2ef3059129b6a0dec765c5eb0b5/bundles/org.eclipse.swt/Eclipse%20SWT/win32/org/eclipse/swt/widgets/Control.java#L3691

The solution is to change the check in `Control`

```
if (this instanceof Browser) {
	// The Edge browser (default in Windows) embeds webView2
	return (getStyle() & SWT.IE) == 0;
}
```

Reply:
Thanks for the pointer! I have developed on a state without that PR, so I missed that usage of the flag. I will update the PR accordingly.

Reply:
Fixed with 075d949

Reply:
My pleasure.

One thing though: checking for `(getStyle() & (SWT.NONE | SWT.EDGE)) != 0` like 1b76c75e2f517c4265ca75df588e540aabbb61c7 does is not good enough because `x & SWT.NONE == 0` which means that `embedsWind32Control` will always return `false`.

## How to test
- Add one `sysout` in `Control::setRedraw`
```java
if (!redraw && isShown && embedsWin32Control()) {
	System.out.println("Embeds"); // add this one
	return;
}
```
- Start the _Runtime Application_ without any special parameters (use the default browser)
- Resize the _Javadoc_ view.

You'll see that the console does **not** print any `Embeds`